googletest/src/gtest.cc

```diff
@@ -3287,8 +3287,16 @@ static void ColoredPrintf(GTestColor color, const char* fmt, ...) {
     return;
   }
 
+#define GTEST_EMIT_ANSI_COLOR_OUTPUT              \
+  printf("\033[0;3%sm", GetAnsiColorCode(color)); \
+  vprintf(fmt, args);                             \
+  printf("\033[m");  // Resets the terminal to default.
+
 #if GTEST_OS_WINDOWS && !GTEST_OS_WINDOWS_MOBILE && \
-    !GTEST_OS_WINDOWS_PHONE && !GTEST_OS_WINDOWS_RT && !GTEST_OS_WINDOWS_MINGW
+    !GTEST_OS_WINDOWS_PHONE && !GTEST_OS_WINDOWS_RT
+# if GTEST_OS_WINDOWS_MINGW
```

Comment:
Having a macro conditionally hide conditional control-flow is a bit tricky to read. Instead, would we #define a constant to either true or false and then use `if (using_mingw && stdout_is_tty)`
Or something similar?

Reply:
Will do. It will be:

```cpp
if (!using_mingw || stdout_is_tty) {
```
as non-MinGW desktop Windows needs to do this unconditionally.

Switching away from the macro will also mean non-MinGW desktop Windows compiling `GTEST_EMIT_ANSI_COLOR_OUTPUT` and `GetAnsiColorCode()` but never using them.